src/Microsoft.Management.Deployment/PackageCatalog.cpp

```diff
@@ -107,70 +108,78 @@ namespace winrt::Microsoft::Management::Deployment::implementation
+                auto catalogPackage = winrt::make_self<wil::details::module_count_wrapper<
+                    winrt::Microsoft::Management::Deployment::implementation::CatalogPackage>>();
+                catalogPackage->Initialize(m_source, match.Package);
 
-                    auto packageMatchFilter = winrt::make_self<wil::details::module_count_wrapper<
-                        winrt::Microsoft::Management::Deployment::implementation::PackageMatchFilter>>();
-                    packageMatchFilter->Initialize(match.MatchCriteria);
+                auto packageMatchFilter = winrt::make_self<wil::details::module_count_wrapper<
+                    winrt::Microsoft::Management::Deployment::implementation::PackageMatchFilter>>();
+                packageMatchFilter->Initialize(match.MatchCriteria);
 
-                    auto matchResult = winrt::make_self<wil::details::module_count_wrapper<
-                        winrt::Microsoft::Management::Deployment::implementation::MatchResult>>();
-                    matchResult->Initialize(*catalogPackage, *packageMatchFilter);
+                auto matchResult = winrt::make_self<wil::details::module_count_wrapper<
+                    winrt::Microsoft::Management::Deployment::implementation::MatchResult>>();
+                matchResult->Initialize(*catalogPackage, *packageMatchFilter);
 
-                    matches.Append(*matchResult);
-                }
-                isTruncated = searchResult.Truncated;
-            }
-            // Exceptions that may occur in the process of executing an arbitrary command
-            catch (const wil::ResultException& re)
-            {
-                hr = re.GetErrorCode();
-            }
-            catch (const winrt::hresult_error& hre)
-            {
-                hr = hre.code();
-            }
-            catch (const ::AppInstaller::Settings::GroupPolicyException&)
-            {
-                // Policy could have changed since server started.
-                hr = APPINSTALLER_CLI_ERROR_BLOCKED_BY_POLICY;
-            }
-            catch (const std::exception&)
-            {
-                hr = APPINSTALLER_CLI_ERROR_COMMAND_FAILED;
-            }
-            catch (...)
-            {
-                hr = APPINSTALLER_CLI_ERROR_COMMAND_FAILED;
+                matches.Append(*matchResult);
             }
+            isTruncated = searchResult.Truncated;
         }
-        auto findPackagesResult = winrt::make_self<wil::details::module_count_wrapper<
-            winrt::Microsoft::Management::Deployment::implementation::FindPackagesResult>>();
-        // TODO: Add search timeout and error code.
-        winrt::Microsoft::Management::Deployment::FindPackagesResultStatus status = FindPackagesResultStatus(hr);
-        findPackagesResult->Initialize(status, isTruncated, matches);
-
-        return *findPackagesResult;
+        // Exceptions that may occur in the process of executing an arbitrary command

```

Comment:
It seems like we should just have a function to handle this:
```
template <typename T>
HRESULT ExecuteWithExceptionTrap(T&& t)
{
  HRESULT result = S_OK;
  try { t(); }
  catch ... // All of these different types
  return result;
}
```
or copy WIL and create our own `WINGET_CATCH_RETURN()` and/or `WINGET_CATCH_STORE(hr)` macro.